frontend/packages/schema-editor/src/components/TypesInspector/TypesInspector.tsx

```diff
@@ -16,14 +14,18 @@ export interface TypesInspectorProps {
 
 export const TypesInspector = ({ schemaItems }: TypesInspectorProps) => {
   const { t } = useTranslation();
-  const dispatch = useDispatch();
-  const { schemaModel, save, selectedTypePointer, setSelectedTypePointer } = useSchemaEditorAppContext();
+  const { schemaModel, save, selectedTypePointer, setSelectedTypePointer, setSelectedNodePointer } = useSchemaEditorAppContext();
+
+  const setSelectedType = (pointer: string) => {
+    setSelectedTypePointer(pointer);
+    setSelectedNodePointer(pointer);
+  }
 
   const handleAddDefinition = (e: MouseEvent) => {
     e.stopPropagation();
     const name = schemaModel.generateUniqueDefinitionName('name');
     const newNode = schemaModel.addFieldType(name);
-    dispatch(setSelectedAndFocusedNode(newNode.pointer));
+    setSelectedType(newNode.pointer);
```

Comment:
Is it possible to write a test that covers this?

Reply:
Yes, but it's out of scope of the task, and the pull request is already quite large. This component will also be rewritten in #11910.